src/operator/gitea.ts

```diff
@@ -1,112 +1,149 @@
-      ]
-      if (podNamespace && podName) {
-        const exec = new k8s.Exec(kc)
-        // Run gitea CLI command to update the gitea oauth group mapping
-        await exec
-          .exec(
-            podNamespace,
-            podName,
-            'gitea',
-            execCommand,
-            null,
-            process.stderr as stream.Writable,
-            process.stdin as stream.Readable,
-            false,
-            (status: k8s.V1Status) => {
-              console.log('Exited with status:')
-              console.log(JSON.stringify(status, null, 2))
-              console.debug('Changed group mapping to: ', teamNamespaceString)
-            },
-          )
-          .catch((error) => {
-            console.debug('Error occurred during exec:', error)
-            throw error
-          })
-      }
-    } else {
-      console.debug('No team namespaces found')
-    }
-  } catch (error) {
-    console.debug(`Error updating IDP group mapping: ${error.message}`)
-    throw error
+// Callbacks
+const secretsAndConfigmapsCallback = async (e: any) => {
+  const { object } = e
+  const { metadata, data } = object
+
+  if (object.kind === 'Secret' && metadata.name === 'gitea-app-operator-secret') {
+    env.giteaPassword = Buffer.from(data.giteaPassword, 'base64').toString()
+    env.oidcClientId = Buffer.from(data.oidcClientId, 'base64').toString()
+    env.oidcClientSecret = Buffer.from(data.oidcClientSecret, 'base64').toString()
+    env.oidcEndpoint = Buffer.from(data.oidcEndpoint, 'base64').toString()
+  } else if (object.kind === 'ConfigMap' && metadata.name === 'gitea-app-operator-cm') {
+    env.hasArgocd = data.hasArgocd === 'true'
+    env.teamConfig = JSON.parse(data.teamConfig)
+    env.teamNames = keys(env.teamConfig).filter((teamName) => teamName !== 'admin')
+    env.domainSuffix = data.domainSuffix
+  } else return
+
+  if (!env.giteaPassword || !env.teamConfig || !env.oidcClientId || !env.oidcClientSecret || !env.oidcEndpoint) {
+    console.info('Missing required variables for Gitea setup/reconfiguration')
+    return
   }
-}
 
-async function runExecCommand() {
-  try {
-    await execGiteaCLICommand('gitea', 'gitea-0')
-  } catch (error) {
-    console.debug('Error could not run exec command', error)
-    console.debug('Retrying in 30 seconds')
-    await new Promise((resolve) => setTimeout(resolve, 30000))
-    console.log('Retrying to run exec command')
-    await runExecCommand()
+  switch (e.type) {
+    case ResourceEventType.Added:
+    case ResourceEventType.Modified: {
+      try {
+        await runSetupGitea()
+      } catch (error) {
+        console.debug(error)
+      }
+      break
+    }
+    default:
+      break
   }
 }
 
+// Operator
 export default class MyOperator extends Operator {
```

Comment:
I would rename this class to GiteaOperator so its more clear that this is an operator for Gitea.

Reply:
Another thingy that is more nice to have and is more my opinion rather than something you need to do:
To make the code more object-oriented and encapsulated, you could consider moving some of these functions into the MyOperator class. Here is my suggestion:
```
export default class MyOperator extends Operator {
  protected async init() {
    // existing init code...
  }

  private async checkAndExecute() {
    // existing checkAndExecute code...
  }

  private async runSetupGitea() {
    // existing runSetupGitea code...
  }

  private async setupGitea() {
    // existing setupGitea code...
  }

  private async setGiteaOIDCConfig() {
    // existing setGiteaOIDCConfig code...
  }

  private async setGiteaGroupMapping(podNamespace: string, podName: string) {
    // existing setGiteaGroupMapping code...
  }
}
```

Reply:
It is not explained in the PR but this change is just to put the functionality of the task into an operator. Reworking of code is not being done for this PR.

Reply:
Thanks for clarification I also had a call with Ferruh about it and we discuss these kind of changes later.